README.md

```diff
@@ -42,15 +42,15 @@ public class Main {
       analysis.setLoggerLevel(Level.TRACE); // Set desired logger level. Level.TRACE provides additional propagation Information
       analysis.initializeAnalysis();
 
-      List<ActionSequence> actionSequences = analysis.findAllSequences();
+      PCMFlowGraph flowGraph = analysis.findFlowGraph();
 
-      List<ActionSequence> propagationResult = analysis.evaluateDataFlows(actionSequences);
-      
-      for(ActionSequence actionSequence : propagationResult) {
-        List<AbstractActionSequenceElement<?>> violations = analysis.queryDataFlow(actionSequence,
-          it -> false // Constraint goes here, return true, if constraint is violated
-        );
-      }
+      PCMFlowGraph propagatedFlowGraph = analysis.evaluateFlowGraph(flowGraph);
```

Comment:
In general: is it still required to return the evaluated flow graph? This originates from the time where we used immutable structures. Now, that the evaluation has side effects and does actually alter the elements within the flow graph, I see no point why there should be a copy of it. Or am I wrong and there is an actual reason to return a "new" propagated flow graph?

Also, we could think of directly calling the evaluation on the flow graph object instead of asking the analysis because it also only calls the evaluate method and we don't need this indirection anymore in my opinion

Reply:
Flow graphs can now be evaluated by their own, starting at commit 6112ad944e9c27d6b10e55ab20df1a31a13fba80.

There are however some pitfalls:
- Copying a `PartialFlowGraph` is discouraged (References need to be updated internally)
- Evaluating a `PartialFlowGraph` standalone is possible in PCM, but not with DFDs
- By proxy, flow graphs are shallow copied, so they reference the same `PartialFlowGraph` after copy

I think the current implementation is pretty accessible, as those pitfalls are abstracted away

Reply:
Moved to #130